src/operator/numpy/np_matmul_op-inl.h

```diff
@@ -0,0 +1,387 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file np_matmul_op-inl.h
+ * \brief Function definition of matrix numpy-compatible matmul operator
+ */
+
+#ifndef MXNET_OPERATOR_NUMPY_NP_MATMUL_OP_INL_H_
+#define MXNET_OPERATOR_NUMPY_NP_MATMUL_OP_INL_H_
+
+#include <mxnet/operator_util.h>
+#include <vector>
+#include <algorithm>
+#include <memory>
+#include "../mxnet_op.h"
+#include "np_tensordot_op-inl.h"
+#include "np_dot-inl.h"
+
+namespace mxnet {
+namespace op {
+
+static bool MatmulNeedBroadcast(const mxnet::TShape& ashape,
+                                const mxnet::TShape& bshape) {
+  bool NotNeed = true;
```

Comment:
For the variable name, we can use the snake_case: https://en.wikipedia.org/wiki/Snake_case. Also, we may try to avoid double negatives by defining
```C++
bool need_bcast = false;
// DO SOMETHING
return need_bcast;
```